modules/core/exported/client.go

```diff
@@ -21,6 +21,15 @@ const (
 	// Localhost is the client type for a localhost client. It is also used as the clientID
 	// for the localhost client.
 	Localhost string = "09-localhost"
+
+	// Active is a status type of a client. An active client is allowed to be used.
+	Active string = "Active"
+
+	// Frozen is a status type of a client. A frozen client is not allowed to be used.
+	Frozen string = "Frozen"
+
+	// Unknown indicates there was an error in determining the status of a client
+	Unknown string = "Unknown"
```

Comment:
Can you elaborate on your concern of using strings? My primary concern of using enums is modularity and value registration

Enums are useful when you know the full range of constants (connection/channel state is a good example) since it provides uniqueness. We cannot know all types of status a client may have. The problem is that every light client developer would need to register their own enums in order to avoid registration conflict. Registering status types seems unnecessarily complicated without a concrete benefit. I think the performance impact of using constants should be pretty low (if any impact).

It would also make maintaining backwards compatibility difficult since we would either need to pre-reserve a selected amount of enums (otherwise we could never make a backwards compatible addition)

I tried using a middle ground solution of using a typed constant 
```
type Status string
```

This changes the function signature to use `exported.Status` instead of `string`, but it makes using the gRPC clunky since we cannot use `exported.Status` as a return type in the gRPC route. Thus we need to cast to/from the exported.Status when returning. Relayers and external third parties would rely on `string` (unless they decided to cast) whereas the code would rely on its typed string. 

I think my preference is continuing to use string. Client types are strings and we use those in identifiers. I don't see the benefit in increasing complexity and potentially creating a divide between what type is used in core IBC vs outside core IBC. 

Thoughts?

Reply:
Thinking more on this, I guess I'm ok with strings.

My concern is there isn't a very clear global view on what a Status means, beyond just `Active` or some other value.

At the code level this is really just acting as a boolean.

I would be ok with even just an enum: `Active` or `Inactive`. And either log the specific issue for inactiveness, or allow it as a specific query on the client state.

But in the absence of that, strings are fine

Reply:
I agree that `"Active"` is arbitrary and is likely something developers are capable of missing. Same with `"Frozen"`

The primary benefit for `Active` to be used not to be used as a boolean is trivial support for a gRPC route, as stated in the linked [issue](https://github.com/cosmos/ibc-go/issues/98):
```
The benefit of using Status() as opposed to IsActive() is trivial support for a Status gRPC route. If the client state interface only supported IsActive then we would likely need to add a Status gRPC route for every light client as opposed to defining it once in 02-client.
```

What about this as the function signature:
```
Status(ctx, clientStore, cdc) exported.Status, string
```
Where exported.Status is an enum of either `Active, Frozen, InActive` and string is any additional information such as "Expired"?

Core IBC only uses the string in error messages and for gRPC support

Reply:
I like this idea! Though I think we just need `Active, Inactive` no? Frozen is a particular type of inactive

Reply:
> I like this idea! Though I think we just need `Active, Inactive` no? Frozen is a particular type of inactive

I believe #141 requires knowing if the client is Frozen? That's the only use case that comes to mind. No strong preference on this decision. Happy to do whichever you think is best

Reply:
Going with `Active, Frozen, Inactive` since misbehaviour/freezing is already a concept in core IBC, it makes sense to have this status type

Reply:
Also going to use protobuf enums. Turns out go doesn't have an enum type. The only way to replicate enums is via a typed constant, but then you have to add a to/from conversion function to get the string representation. Using proto enums seems like the most sensible option

Reply:
I decided to use a typed string. Initially I implemented a typed int32 and the function signature above (hence the force push), but I realized separating the status and info felt unnecessary. The main critique in this thread is separating a string from a status and a typed string accomplishes that. 

It also is self descriptive. The value `1` indicates `Frozen` a lot less than the string `"Frozen"`. Furthermore, backwards compatibility will be easier since the next value added wouldn't have to be `3`. 

Since we are using a typed string, I kept `Active, Frozen, Expired, Unknown` as the default supported status types. Core IBC should only ever look for `Active` and potentially `Frozen` in exceptional cases. 

I tried using proto enums but proto enums will just create a typed int32 and requires the active status to be suffixed with `_UNSPECIFIED` per proto linting. In the gRPC response, since I couldn't use a typed Status, I went with string as I figured relayers would prefer to compare against a self descriptive string rather than an arbitrary number. 

I think there are also less risks for errors, since core IBC defensively uses the status (must be Active to do anything). If I used a typed int32, a typed status return value of 0 would always be interpreted as Active by core IBC. For example
```
var inactive exported.Status = 0
return inactive
```

With a typed string, returning a typed status of `"Active"` is self descriptive. You wouldn't use that return value to indicate the client is inactive. 

This last part is a minor nit, but a typed int32 just felt like unnecessary indirection.